Sources/Testing/Events/Recorder/Event.AdvancedConsoleOutputRecorder.swift

```diff
@@ -0,0 +1,79 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+extension Event {
+  /// An experimental console output recorder that provides enhanced test result
+  /// display capabilities.
+  ///
+  /// This recorder is currently experimental and must be enabled via the
+  /// `SWT_ENABLE_EXPERIMENTAL_CONSOLE_OUTPUT` environment variable.
+  ///
+  /// Future capabilities will include:
+  /// - Hierarchical test result display with tree visualization
+  /// - Live progress indicators during test execution
+  /// - Enhanced SF Symbols integration on supported platforms
+  struct AdvancedConsoleOutputRecorder: Sendable {
+    /// Configuration options for the advanced console output recorder.
+    struct Options: Sendable {
+      /// Base console output recorder options to inherit from.
+      var base: Event.ConsoleOutputRecorder.Options
+      
+      /// Whether to enable experimental hierarchical output display.
+      /// Currently unused - reserved for future PR #2.
+      var useHierarchicalOutput: Bool
+      
+      /// Whether to show successful tests in the output.
+      /// Currently unused - reserved for future PR #2.
+      var showSuccessfulTests: Bool
+      
+      init() {
+        self.base = Event.ConsoleOutputRecorder.Options()
+        self.useHierarchicalOutput = true
+        self.showSuccessfulTests = true
+      }
+    }
+    
+    /// The options for this recorder.
+    let options: Options
+    
+    /// The write function for this recorder.
+    let write: @Sendable (String) -> Void
+    
+    /// The fallback console recorder for standard output.
+    private let _fallbackRecorder: Event.ConsoleOutputRecorder
+    
+    /// Initialize the advanced console output recorder.
+    ///
+    /// - Parameters:
+    ///   - options: Configuration options for the recorder.
+    ///   - write: A closure that writes output to its destination.
+    init(options: Options = Options(), writingUsing write: @escaping @Sendable (String) -> Void) {
+      self.options = options
+      self.write = write
+      self._fallbackRecorder = Event.ConsoleOutputRecorder(options: options.base, writingUsing: write)
+    }
+  }
+}
+
+extension Event.AdvancedConsoleOutputRecorder {
+  /// Handle an event by processing it and generating appropriate output.
+  ///
+  /// Currently this is a skeleton implementation that delegates to the
+  /// standard ConsoleOutputRecorder.
+  ///
+  /// - Parameters:
+  ///   - event: The event to handle.
+  ///   - eventContext: The context associated with the event.
+  func handle(_ event: borrowing Event, in eventContext: borrowing Event.Context) {
```

Comment:
Why `handle()` instead of `record()`? `record()` would be consistent with the existing interfaces for other types in this folder.